provider/datagen/src/cldr/transform/locale_canonicalizer/likely_subtags.rs

```diff
@@ -72,62 +72,77 @@ impl IterableResourceProvider<LikelySubtagsV1Marker> for LikelySubtagsProvider {
     }
 }
 
-impl From<&cldr_serde::likely_subtags::Resource> for LikelySubtagsV1 {
+impl From<&cldr_serde::likely_subtags::Resource> for LikelySubtagsV1<'static> {
     fn from(other: &cldr_serde::likely_subtags::Resource) -> Self {
-        use icu_locid::LanguageIdentifier;
-
-        let mut language_script = LiteMap::new();
-        let mut language_region = LiteMap::new();
-        let mut language = LiteMap::new();
-        let mut script_region = LiteMap::new();
-        let mut script = LiteMap::new();
-        let mut region = LiteMap::new();
-        let mut und = LanguageIdentifier::default();
-
-        // Create a result LanguageIdentifier. We only need to store the delta
-        // between the search LanguageIdentifier and the result LanguageIdentifier.
-        let extract_result =
-            |entry: &(LanguageIdentifier, LanguageIdentifier)| -> LanguageIdentifier {
-                LanguageIdentifier {
-                    language: if entry.0.language != entry.1.language {
-                        entry.1.language
-                    } else {
-                        icu_locid::subtags::Language::UND
-                    },
-                    script: if entry.0.script != entry.1.script {
-                        entry.1.script
-                    } else {
-                        None
-                    },
-                    region: if entry.0.region != entry.1.region {
-                        entry.1.region
-                    } else {
-                        None
-                    },
-                    variants: icu_locid::subtags::Variants::default(),
-                }
-            };
+        use icu_locid::subtags::Language;
+
+        let mut language_script = ZeroMap::new();
+        let mut language_region = ZeroMap::new();
+        let mut language = ZeroMap::new();
+        let mut script_region = ZeroMap::new();
+        let mut script = ZeroMap::new();
+        let mut region = ZeroMap::new();
+        let mut und = None;
 
         for entry in other.supplemental.likely_subtags.iter() {
+            // Computes the delta of the entry and assigns to the pattern.
+            // Errors if the delta is not assignable to the pattern.
+            macro_rules! with_diff {
+                ($pat:pat => $stmt:expr ) => {
+                    if let $pat = (
+                        if entry.0.language != entry.1.language {
+                            entry.1.language
+                        } else {
+                            Language::UND
```

Comment:
`question`: why `UND` instead of `Option<Language>` and None here?

Reply:
Because `UND` is our (and BCP-47's) representation of an absent language code. If we used `Option<Language>`, absence would have two values (`Some(Language::UND)` and `None`).

Reply:
This is consistent with the three (stack) fields of `LanguageIdentifier`

Reply:
The reason we do this on `LanguageIdentifier` is that LI *cannot* exist without language. The "empty" language is therefore just one of the languages rather than representation of lack of language.
It is intended to help API users understand the difference between omitting script/region and omitting language.

In your code that is not important. You navigate on the LSR triplet and communicate which of the three are present. I think `(Option<Language>, Option<Script>, Option<Region>)` is the correct type for it.

Reply:
Ah, that's a good point! 
Thank you for taking time to explain it to me. I'm convinced.